File: 2022-2023/sourcecode/lab/lab2/ExplicitIntentJava/app/src/main/java/pl/edu/uwr/pum/explicitintentjava/SecondActivity.java
package pl.edu.uwr.pum.explicitintentjava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    public static final String EXTRA_REPLY = "pl.edu.uwr.pum.explicitintentjava.REPLY";

    private EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        editText = findViewById(R.id.editTextSecond);

        Intent intent = getIntent();
        String message = intent.getStringExtra(MainActivity.EXTRA_MESSAGE);
        TextView textView = findViewById(R.id.textView);
        textView.setText(message);
    }

    public void returnMessage(View view) {
        String reply = editText.getText().toString();
        Intent intent = new Intent();

        intent.putExtra(EXTRA_REPLY, reply);
        setResult(RESULT_OK, intent);
        finish();
    }
}

File: 2022-2023/sourcecode/lab/lab2/ExplicitIntentKotlin/app/src/main/java/pl/edu/uwr/pum/explicitintentkotlin/SecondActivity.kt
package pl.edu.uwr.pum.explicitintentkotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView

const val EXTRA_REPLY = "pl.edu.uwr.pum.explicitintentkotlin.REPLY"

class SecondActivity : AppCompatActivity() {

    private val editText: EditText by lazy { findViewById(R.id.editTextSecond) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val message = intent.getStringExtra(EXTRA_MESSAGE)
        findViewById<TextView>(R.id.textView).apply {
            text = message
        }
    }

    fun returnMessage(view: View) {
        setResult(
            RESULT_OK,
            Intent().apply {
                putExtra(EXTRA_REPLY,
                    findViewById<EditText>(R.id.editTextSecond).text.toString())
            })
        finish()
    }
}